2018/002-update-channels.md

```diff
@@ -0,0 +1,123 @@
+ Title
+
+| Field  | Value      |
+|:-------|:-----------|
+| Status | Draft      |
+| Date   | 2018-07-31 |
+
+## Introduction
+
+### Problem description
+
+Automatic updates can be disruptive for a customer's application stack (ex.
+Kubernetes minor upgrade, default pod security policy introduction).
+
+Right now there is no visibility of what is inside an update and furthermore
+there is no possible selection of which updates are applied. An admin needs to
+apply all updates that come through a single channel.
+
+In order to selectively apply updates in SUSE CaaS Platform, we need different
+update channels. This allows the separation of possibly disruptive updates
+into their own channels, and lets the admin choose a conservative or progressive
+update mechanism. The admin can also see what is inside an update to be able
+to decide when to apply it.
+
+### Proposed change
+
+The admin will have the ability of choosing when to switch to the next channel.
+A link to the release notes of the update channel is displayed and can be
+evaluated by the admin beforehand.
+
+The UI of velum needs to be adapted to enable a switch to a new channel.
+
+## Detailed RFC
+
+We can separate the updates into different channels. There are 2 types of channels:
+
+* maintenance channel: normal maintenance updates, which are safe and non-intrusive
+* new version channel: version updates which can contain intrusive updates
+
+This RFC proposes:
+
+* a front-end change in the velum UI to enable the admin to switch to a new version
+channel
+* a back-end change in salt to perform the channel switch on all cluster nodes
+
+### Proposed change (Detailed)
+
+#### Tools/Services
+
+We need a set of tools/services to indicate, whenever a maintenance update or a new version
+update is ready to be installed. They should run on each worker node to be more independent
+and decoupled from velum. If velum would be the initiator of the update check, it might lead
+to inconsistent and long execution runs in larger clusters.
+
+* "check for updates" service on each worker
+  - reports availability of maintenance updates to velum (via grains?)
+  - once per day e.g.
+  - decoupled from velum, runs independently, just reports
+* "new version" checker service on each worker
+  - check for new versions (migration targets) (via SCC? or own solution)
```

Comment:
RMT/SMT/SCC provide an API which we query, if there is a new compatible version available for this CaaSP version. We will use the SUSEConnect library, no need to re-implement this.